### PR TITLE
[#3884] fix(common): fix type cast issue in `DTOConverters`

### PR DESCRIPTION
**Pull request by xiaozcy**
### What changes were proposed in this pull request?

fix type cast issue in `DTOConverters`.

### Why are the changes needed?

Fix: #3884 

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Not need.

### CODE REVIEW

File: common/src/main/java/com/datastrato/gravitino/dto/util/DTOConverters.java [hunk 1]
```diff
@@ -316,13 +316,19 @@ public static Partitioning toDTO(Transform transform) {
 
     } else if (transform instanceof Transforms.ListTransform) {
       Transforms.ListTransform listTransform = (Transforms.ListTransform) transform;
-      return ListPartitioningDTO.of(
-          listTransform.fieldNames(), (ListPartitionDTO[]) toDTOs(listTransform.assignments()));
+      ListPartitionDTO[] assignments =
+          Arrays.stream(listTransform.assignments())
+              .map(DTOConverters::toDTO)
+              .toArray(ListPartitionDTO[]::new);
+      return ListPartitioningDTO.of(listTransform.fieldNames(), assignments);
```
Comment (mchades) [inline common/src/main/java/com/datastrato/gravitino/dto/util/DTOConverters.java:323]: What would be the problem if we use the way we forced the conversion before the change? Can we add tests to cover this change?

File: common/src/test/java/com/datastrato/gravitino/dto/util/TestDTOConverters.java [hunk 2]
```diff
@@ -181,4 +185,42 @@ void testListPartitionConvert() {
...
     Assertions.assertEquals(properties, listPartitionDTO.properties());
   }
+
+  @Test
+  void testPartitionDTOConvert() {
+    Literal<?> lower = Literals.stringLiteral("2008-08-08");
+    Literal<?> upper = Literals.stringLiteral("us");
+    Map<String, String> properties = Collections.singletonMap("key", "value");
+    RangePartition rangePartition =
+        (RangePartition) Partitions.range("range", upper, lower, properties);
```
Comment (mchades) [inline common/src/test/java/com/datastrato/gravitino/dto/util/TestDTOConverters.java:195]: Can we modify the method `Partitions.range` method to directly return `RangePartition`?

Comment (xiaozcy) [reply common/src/test/java/com/datastrato/gravitino/dto/util/TestDTOConverters.java:195]: Looks good to me, I'll give it a try.

Comment (xiaozcy) [reply common/src/test/java/com/datastrato/gravitino/dto/util/TestDTOConverters.java]: This is already done.

**Review by mchades — APPROVED** [review]
LGTM, thanks for your contributions!

**Pull request merged**